pkg/devfile/adapters/common/command.go

```diff
@@ -107,6 +107,19 @@ func validateAction(data data.DevfileData, action common.DevfileCommandAction) (
 	return
 }
 
+// GetInitCommand iterates through the components in the devfile and returns the init command
+func GetInitCommand(data data.DevfileData, devfileInitCmd string) (initCommand common.DevfileCommand, err error) {
+	if devfileInitCmd != "" {
+		// a init command was specified so if it is not found then it is an error
+		initCommand, err = getCommand(data, devfileInitCmd, true)
+	} else {
+		// a init command was not specified so if it is not found then it is not an error
+		initCommand, err = getCommand(data, string(DefaultDevfileInitCommand), false)
+	}
+
+	return
```

Comment:
you don't check for `err` and you return just `return`.

You should be doing `return err`

Reply:
Since this function uses a "named" return value of `(initCommand common.DevfileCommand, err error)`, the function will return those two variables. Within the function, these two variables will be set by either calls to the `getCommand`. We will be simply returning to same values back after they have been set.  https://golang.org/ref/spec#Return_statements